climada/engine/uncertainty/base.py

```diff
@@ -0,0 +1,771 @@
+            imp_fun.paa = np.array([sigmoid_function(v, G, v_half, vmin, k)
+                                    for v in imp_fun.intensity])
+            imp_fun.check()
+            impf_set = ImpactFuncSet()
+            impf_set.append(imp_fun)
+            return impf_set
+        distr_dict = {"G": sp.stats.uniform(0.8, 1),
+              "v_half": sp.stats.uniform(50, 100),
+              "vmin": sp.stats.norm(loc=15, scale=30),
+              "k": sp.stats.randint(low=1, high=9)
+              }
+        unc_var_cont = UncVar(imp_fun_tc, distr_dict)
+
+    """
+
+    def __init__(self, uncvar_func, distr_dict):
+        """
+        Initialize UncVar
+
+        Parameters
+        ----------
+        uncvar_func : function
+            Variable defined as a function of the uncertainty parameters
+        distr_dict : dict
+            Dictionary of the probability density distributions of the
+            uncertainty parameters, with keys matching the keyword
+            arguments (i.e. uncertainty parameters) of the uncvar_func
+            function.
+            The distribution must be of type scipy.stats
+            https://docs.scipy.org/doc/scipy/reference/stats.html
+
+        Returns
+        -------
+        None.
+
+        """
+        self.labels = list(distr_dict.keys())
+        self.distr_dict = distr_dict
+        self.uncvar_func = uncvar_func
+
+
+    def plot(self):
+        """
+        Plot the distributions of the parameters of the uncertainty variable.
+
+        Returns
+        -------
+        fig, axes: matplotlib.pyplot.figure, matplotlib.pyplot.axes
+            The figure and axis handle of the plot.
+
+        """
+
+        nplots = len(self.distr_dict)
+        nrows, ncols = int(np.ceil(nplots / 3)), min(nplots, 3)
+        fig, axes = plt.subplots(nrows=nrows, ncols=ncols, figsize=(20, 16))
+        if nplots > 1:
+            flat_axes = axes.flatten()
+        else:
+            flat_axes = np.array([axes])
+        for ax, name_distr in zip_longest(flat_axes,
+                                    self.distr_dict.items(), 
+                                    fillvalue=None):
+            if name_distr is None:
+                ax.remove()
+                continue
+            (param_name, distr) = name_distr
+            x = np.linspace(distr.ppf(0.001), distr.ppf(0.999), 100)
+            ax.plot(x, distr.pdf(x), label=param_name)
+            ax.legend()
+        return fig, axes
+
+
+    def evaluate(self, uncvar_kwargs):
+        """
+        Evaluate the uncertainty variable.
+
+        Parameters
+        ----------
+        uncvar_kwargs : dict
+            These parameters will be passed to self.uncvar_func.
+            They must be the input parameters of the uncertainty variable .
+
+        Returns
+        -------
+
+            Evaluated uncertainty variable
+
+        """
+        return self.uncvar_func(**uncvar_kwargs)
+    
```

Comment:
trailing space - also in many other places...

Reply:
Thanks, will remove them!